torch/ao/pruning/_experimental/pruner/README.md

```diff
@@ -1,93 +1,263 @@
+
+It is up to the user to define the following two items.
+1. What layers are you looking to prune
+2. Your pruning criteria - basically how you pick weights to prune and to what degree.
+
+## Structured Pruning for Supported Modules
+Here is an example script that will prune away 50% of the rows for all the linear layers in the model, based on the saliency of each row.
+```python
+from torch.ao.pruning._experimental.pruner import BaseStructuredSparsifier
+from torch.ao.pruning._experimental.base_structured_sparsifier import _get_default_structured_pruning_patterns
+#define model
+class Model(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.seq = nn.Sequential(
+            nn.Linear(700, 500, bias=True), # Alpha
+            nn.ReLU(),
+            nn.Linear(500, 800, bias=False), # Beta
+            nn.ReLU(),
+            nn.Linear(800, 600, bias=True), # Gamma
+            nn.ReLU(),
+        )
+        self.linear = nn.Linear(600, 4, bias=False) # Delta
+
+    def forward(self, x):
+        x = self.seq(x)
+        x = self.linear(x)
+        return x
+
+# define config, specify fqn of layers you want to prune and target sparsity level
+pruning_config = [
+    {"tensor_fqn": "seq.0.weight", "sparsity_level": 0.5},
+    {"tensor_fqn": "seq.2.weight", "sparsity_level": 0.5},
+    {"tensor_fqn": "seq.4.weight", "sparsity_level": 0.5},
+    {"tensor_fqn": "linear.weight", "sparsity_level": 0.5},
+]
+
+# define pruner
+class SaliencyPruner(BaseStructuredSparsifier):
+     """
+     Prune filters based on the saliency
+     The saliency for a filter is given by the sum of the L1 norms of all of its weights
+     """
+
+     def update_mask(self, module, tensor_name, **kwargs):
+        # tensor_name will give you the FQN, all other entries in sparse config is present in kwargs
+         weights = getattr(module, tensor_name)
+         mask = getattr(module.parametrizations, tensor_name)[0].mask
+
+         # use negative weights so we can use topk (we prune out the smallest)
+         saliency = -weights.norm(dim=tuple(range(1, weights.dim())), p=1)
+         num_to_pick = int(len(mask) * kwargs["sparsity_level"])
+         prune = saliency.topk(num_to_pick).indices
+
+         # Set the mask to be false for the rows we want to prune
+         mask.data[prune] = False
+
+# define patterns
+patterns = _get_default_structured_pruning_patterns()
+
+original = Model()
+# any configs passed in here are defaults that are propagated
+pruner = SaliencyPruner({"prune_bias": True}, patterns=patterns)
+pruner.prepare(original, sparse_config)
+pruner.enable_mask_update = True
+pruner.step()
+pruned_model = pruner.prune()
+```
 
-The Base Pruner inherits from the Base Sparsifier.
+Let's walk through this line by line to understand what the different parts are doing.
```

Comment:
may be better to walk line by line with comments in the code block, as is, this section seems less line-by-line and more just going over the main parts of the APII